mitodl/mitxonline
# add MM program certificate migration script

#### Pre-Flight checklist


#### What are the relevant tickets?
https://github.com/mitodl/mitxonline/issues/706

#### What's this PR do?
Adds a sql script to migrate learners' program certificates, especially DEDP from MicroMasters, it also adds a field to the ProgramId model for program certificate revision because querying program certificate revision is cumbersome (This field will be manually added to RC and production)

#### How should this be manually tested?

- Ensure your local username exists on [MicroMaster RC](https://micromasters-rc.odl.mit.edu/admin/social_django/usersocialauth/) (mitxonline provider)
- DEDP program and program certificate are set up in CMS. see https://rc.mitxonline.mit.edu/cms/pages/74/edit/ for reference
- (optional) at least one [course run certificate](http://mitxonline.odl.local:8013/admin/courses/courseruncertificate/) for the DEDP program - this is used for calculating the start/end date displayed on users' program certificate
- Follow the [README](https://github.com/mitodl/mitxonline/tree/main/micromasters_import#local-setup) to set up a local foreign data wrapper mounted to MicroMaster RC if haven't already
- Run `./manage.py migrate micromasters_import` to add program_certificate_revision field
- Go to http://mitxonline.odl.local:8013/admin/micromasters_import/programid/ to enter the DEDP program mapping
(program certificate revision can be queried like `SELECT MIN(id) FROM wagtailcore_pagerevision WHERE page_id = <PROGRAM CERTIFICATE PAGE ID>` to get the oldest version)
e.g.
![image](https://user-images.githubusercontent.com/3138890/206266738-83bc6b02-98fc-4b7f-948e-e88de2928645.png)
- run migration script:  `./manage.py import_micromasters_data --num 012`
- check the imported program certificates http://mitxonline.odl.local:8013/admin/courses/programcertificate/
- to view a program certificate: http://mitxonline.odl.local:8013/certificate/program/(UUID)/

![image](https://user-images.githubusercontent.com/3138890/206269042-82816b52-c4fa-4be5-8938-e83f83ed6530.png)
